Makefile.common

```diff
@@ -120,3 +118,26 @@ ifneq ($(EXE),)
 $(1): $(1)$(EXE)
 endif
 endef # PROGRAM_SYNONYM
+
+# Lexer generation
+
+BOOT_OCAMLLEX ?= $(CAMLRUN) $(ROOTDIR)/boot/ocamllex
+
+# The OCAMLLEX command use in the recipe below is define in the individual
+# makefiles, because its default value is directory-specific.
+# Some Directories use BOOT_OCAMLLEX, others use the freshly compiled
+# lexer generator.
+
+OCAMLLEXFLAGS ?= -q
+
+%.ml: %.mll
+	$(OCAMLLEX) $(OCAMLLEXFLAGS) $<
+
+# Parser generation
+
+OCAMLYACC ?= $(ROOTDIR)/yacc/ocamlyacc$(EXE)
+
+OCAMLYACCFLAGS ?=
+
+%.ml %.mli: %.mly
+	$(OCAMLYACC) $(OCAMLYACCFLAGS) $<
```

Comment:
A question after a quick look, how is the menhir-generated compiler parser in `parsing/parser.mly` handled?